snuba/query/conditions.py

```diff
@@ -188,6 +191,27 @@ def _get_first_level_conditions(
     In the AST, the condition is a tree, so we need some additional
     logic to extract the operands of the top level AND condition.
     """
+    # Ignore the identity pattern, e.g. cond = 1 or 1 = cond since Sentry uses this
+    # format frequently (in Discover) as a workaround to current schema limitations.
+    nested_func_call = Param("nested_func", FunctionCallPattern())
+    lit = LiteralPattern(None, Int(1))
+
+    id_pattern = Or(
+        [
+            FunctionCallPattern(
+                None, String(ConditionFunctions.EQ), (nested_func_call, lit),
+            ),
+            FunctionCallPattern(
+                None, String(ConditionFunctions.EQ), (lit, nested_func_call),
+            ),
+        ]
+    )
+
+    match = id_pattern.match(condition)
+
+    if match:
+        condition = match.expression("nested_func")
```

Comment:
This method is used in several key parts of query processing, like prewhere and project_id finder. Before making a functional change we need to ensure we are not breaking anything.
First I would suggest you run all sentry tests against snuba.
Also what about doing a dry run of this: match the condition and log if the result ids going to be different from the old implementation. In this case if nested_func.function_name is AND or OR, the result will be different.
Log these discrepancies in sentry, then we can have some confidence whether we are breaking anything.